src/base/gfx/pathsampler.cpp

```diff
@@ -21,32 +21,30 @@ void PathSampler::path(const Geom::Point &pConv0,
     double i1,
     size_t recurseCnt)
 {
-	const size_t maxRecursion = 20;
+	constexpr static size_t maxRecursion = 20;
 	if (recurseCnt >= maxRecursion) return;
 
-	auto i = (i0 + i1) / 2.0;
-	auto pConv = getPoint(i);
+	const auto sqrLength = (pConv1 - pConv0).sqrAbs();
+	if (Math::Floating::is_zero(sqrLength)) return;
 
-	const Geom::Triangle triangle{{pConv0, pConv, pConv1}};
-	auto area = triangle.area();
-	auto height = 2 * area / (pConv1 - pConv0).abs();
```

Comment:
The test case failures are caused by instances where the division divisor is '0'.